Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial:IEquatable<Polynomial>
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length>100)
+            {
+                throw new Exception("Max power must be less equal 100");
+            }
+            _coefficients = coefficients;
+        }
+
+        public bool Equals(Polynomial? other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (!(other is Polynomial || this is Polynomial) )
+            {
+                return false;
+            }
+
+            if (this._coefficients.Length != other._coefficients.Length)
+            {
+                return false;
+            }
+
+            if (this == other)
```

Comment:
You can merge that two last returns as 
`return this == other;`

Reply:
And I am sure that resharper gives you that suggestion 😉

Reply:
Yeah, it did, I was not sure whether I want it or not.  Swapped.